### PR TITLE
Take CSRF token from csrfmiddlewaretoken field instead of from cookie

### PR DESCRIPTION
**Pull request by tadeo**
Allows to run with setting CSRF_COOKIE_HTTPONLY = True
Fixes Issue #70

### CODE REVIEW

**Comment by bradleyg** [comment]
Thanks for this. Any way we can fallback to reading it from the DOM only if it's missing from the cookie? The problem is that some existing users might not be currently outputting the CSRF token into the HTML.

File: s3direct/static/s3direct/js/scripts.js [hunk 1]
```diff
@@ -131,7 +125,7 @@
             dest     = el.querySelector('.file-dest').value,
             url      = el.getAttribute('data-policy-url'),
             form     = new FormData(),
-            headers  = {'X-CSRFToken': getCookie('csrftoken')}
+            headers  = {'X-CSRFToken': document.querySelector('input[name=csrfmiddlewaretoken]').value}
```
Comment (bradleyg) [inline s3direct/static/s3direct/js/scripts.js]: Maybe:
```
headers  = {
  'X-CSRFToken': getCookie('csrftoken') || document.querySelector('input[name=csrfmiddlewaretoken]').value
}
```

**Comment by tadeo** [comment]
Thanks for considering this PR. It's nice from your side to consider existing users, I will advice them through the readme or release notes to update to a more standard way of using CSRF in Django (explicitly adding the token in the forms). What do you think?

**Comment by jwilliams-ecometrica** [comment]
@bradleyg , @tadeo , 

Has there been any progress on this PR?

**Comment by tadeo** [comment]
Thanks @jwilliams-ecometrica for taking this back to life. As I was uploading large video files in my project I switched to FileUploader so I forgot this PR.
@bradleyg I took your recommendation of supporting existing users and I take the csrfmiddlewaretoken and if it's not present I fall back to the cookie.

**Comment by jwilliams-ecometrica** [comment]
@tadeo Thanks for responding so quickly!

**Comment by tadeo** [comment]
@bradleyg thanks to @jwilliams-ecometrica question I revisited this PR. I updated it to match current master branch and extended the support to the AWS multipart upload.
Feel free to review it and make any comments, thanks a lot!

**Comment by bradleyg** [comment]
Nice! Thanks for this, haven't had time to review/merge it, but I'll get round to it this week.

**Pull request merged**